src/sql-parser/src/ast/defs/ddl.rs

```diff
@@ -136,27 +136,75 @@ impl<T: AstInfo> AstDisplay for ProtobufSchema<T> {
 impl_display_t!(ProtobufSchema);
 
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub enum CsrConfigOptionName {
+pub enum CsrConfigOptionName<T: AstInfo> {
     AvroKeyFullname,
     AvroValueFullname,
     NullDefaults,
+    AvroDocOn(AvroDocOn<T>),
+}
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct AvroDocOn<T: AstInfo> {
+    pub identifier: DocOnIdentifier<T>,
+    pub for_schema: DocOnSchema,
+}
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum DocOnSchema {
+    KeyOnly,
+    ValueOnly,
+    All,
 }
 
-impl AstDisplay for CsrConfigOptionName {
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum DocOnIdentifier<T: AstInfo> {
+    Column {
+        item_name: T::ItemName,
+        column_name: Ident,
```

Comment:
OOC will it be a good idea or a bad idea to have something like T::SubItemName which can automatically resolve to a column name?

Reply:
I think that'd be a great idea! Might make a good followup PR so that `CommentObjectType::Column` can get the same treatment.

Reply:
https://materializeinc.slack.com/archives/C02FWJ94HME/p1696540204393639

Reply:
Cool, I will try this as a follow up 👍

Reply:
Addressed in https://github.com/MaterializeInc/materialize/pull/22244